resources-domain/packages/resources-search-config/convertIndexDocument.ts

```diff
@@ -64,12 +64,15 @@ export const convertIndexDocument = (
     key: string,
     attribute: ReferrableResourceAttribute<boolean | string | number>,
   ) => {
-    const fieldAndMapping = getMappingField(resourceIndexDocumentMappings, key);
-    if (fieldAndMapping) {
-      return pushValueToMappedField(
-        fieldAndMapping,
-        fieldAndMapping.mapping.indexValueGenerator?.(attribute) ?? attribute.value,
-      );
+    const fieldAndMappings = getMappingFields(resourceIndexDocumentMappings, key);
+    if (fieldAndMappings.length) {
+      for (const fieldAndMapping of fieldAndMappings) {
+        pushValueToMappedField(
+          fieldAndMapping,
+          fieldAndMapping.mapping.indexValueGenerator?.(attribute) ?? attribute.value,
+        );
+      }
+      return;
```

Comment:
I'm personally not a huge fan of `if (fieldAndMappings.length) { ... }` because it relies on the fact that the integer `0` will be evaluated to `false`. For more clarity, I prefer the explicit comparison `if (fieldAndMappings.length > 0) { ... }`.

But it's up to you to change it or not.